node/core/approval-voting/src/lib.rs

```diff
@@ -1925,14 +2006,13 @@ fn process_wakeup(
 async fn launch_approval(
 	ctx: &mut impl SubsystemContext,
 	metrics: Metrics,
-	mut background_tx: mpsc::Sender<BackgroundRequest>,
 	session_index: SessionIndex,
 	candidate: &CandidateReceipt,
 	validator_index: ValidatorIndex,
 	block_hash: Hash,
 	candidate_index: usize,
 	backing_group: GroupIndex,
-) -> SubsystemResult<Option<RemoteHandle<()>>> {
+) -> SubsystemResult<Option<Timeout<RemoteHandle<ApprovalState>>>> {
```

Comment:
As per the comment in `CurrentlyCheckingSet` - `Timeout` just returns `None` if it is triggered, so we should have this function return only the `RemoteHandle` and then have `CurrentlyCheckingSet` wrap it in something like this:

```rust
async move {
    match remote_handle.timeout(..).await {
         None => ApprovalState { candidate_hash, TimedOut },
         Some(approval_output) => ApprovalState { candidate_hash, approval_output},
    }
}
```